lib/storage/table.go

```diff
@@ -267,22 +267,20 @@ func (tb *table) MustAddRows(rows []rawRow) {
 	if len(rows) == 0 {
 		return
 	}
-
+	// Get the maximum and minimum timestamps in the rows
+	maxTsInRows, minTsInRows := rows[0].Timestamp, rows[0].Timestamp
+	for i := range rows {
```

Comment:
The code in this pull request appears to be more concise. The original one has a chance to break earlier and probably doesn't need to go through all the rows. Any thought on this? @f41gh7

Reply:
Break earlier seems to be a rare case for cross partition data ingestion.

Currently, data partitioned by month. And it should be a rare case, when a single data block has timestamps for multiple months. 

 So for me it looks like valid optimization. Most common case is single partition data ingestion, it'll not trigger previous version fast path.

 Also It'd be great to have a bench for it.